source/core/solid_base.cc

```diff
@@ -167,6 +174,79 @@ SolidBase<dim, spacedim>::get_solid_velocity()
   return velocity;
 }
 
+template <int dim, int spacedim>
+void
+SolidBase<dim, spacedim>::integrate_velocity(double time_step)
+{
+  for (auto particle = solid_particle_handler->begin();
+       particle != solid_particle_handler->end();
+       ++particle)
+    {
+      Point<spacedim> particle_location = particle->get_location();
+      for (unsigned int comp_i = 0; comp_i < spacedim; ++comp_i)
+        {
+          double k1 = velocity->value(particle_location, comp_i);
+
+          Point<spacedim> p1 = particle_location;
+          p1[comp_i] += time_step / 2 * k1;
+          double k2 = velocity->value(p1, comp_i);
+
+          Point<spacedim> p2 = particle_location;
+          p2[comp_i] += time_step / 2 * k2;
+          double k3 = velocity->value(p2, comp_i);
+
+          Point<spacedim> p3 = particle_location;
+          p3[comp_i] += time_step * k3;
+          double k4 = velocity->value(p3, comp_i);
+
+          particle_location[comp_i] +=
+            time_step / 6 * (k1 + 2 * k2 + 2 * k3 + k4);
+        }
+      particle->set_location(particle_location);
```

Comment:
There is a mistake here. When comp_i=0 it is correct, but when comp_i=1, then you have already updated the particle location of comp_i=0. This means that you are taking the new x position when calculating your velocity. 
The solution is to calculate the velocity in a loop, but to do the operation on the points directly. k1,k2,k3, k4 should be of type tensor<1,dim>